retryrpc/api.go

```diff
@@ -0,0 +1,134 @@
+// Package retryrpc provides a client and server RPC model which survives
+// lost connections on either the client or the server.
+package retryrpc
+
+import (
+	"container/list"
+	"sync"
+	"time"
+)
+
+// Request is the structure sent
+type Request struct {
```

Comment:
So by this type (Request) and the corresponding type Response, we are jettisoning actual JSON RPC, correct? Sure, we've decided to marshal via JSON... but we no longer need anything like the form that JSON RPC defines namely:
```
type jrpcRequestStruct struct {
	JSONrpc string         `json:"jsonrpc"`
	Method  string         `json:"method"`
	ID      uint64         `json:"id"`
	Params  [1]interface{} `json:"params"`
}

type jrpcResponseWithErrorStruct struct {
	ID     uint64      `json:"id"`
	Error  string      `json:"error"`
	Result interface{} `json:"result"`
}
```
Correct?

Reply:
Request.JReq will contain JSON with these fields:

    type jsonBypassRequest struct {
    	    Method  string         `json:"method"`
	    Params  [1]interface{} `json:"params"`
    }

and the Response.Res will contain JSON with these fields:

    typeBypassResponse struct {
             Result interface{}  `json:"result"`
    }

I will not be using jrpcRequestStruct or jrpcResponseWithErrorStruct.

Did that answer your question?

I will update the PR with these responses to include the marshaling and unmarshaling.

Reply:
Yeah, as I mentioned verbally, I doubt we need the whole JSON-RPC doc in there...